vltt/assignment3/solution.cpp

```diff
@@ -0,0 +1,197 @@
+#define _SILENCE_TR1_NAMESPACE_DEPRECATION_WARNING
```

Comment:
[optional] Trie is the optimal implementation for a dictionary, however you're not taking full advantage of the data structure :)

For each `passage_through_the_grid` call you look up prefix/word from the root, which is inefficient, up to O(N) when you look up the whole word. You can improve this lookup time to O(1), by passing around the current node into the recursive call, such that you only have to preform one lookup in the subtree you haven't explored yet. 

Then you wouldn't need to pass the whole prefix either, just the next character.